pkg/controllers/vdb/podfacts.go

```diff
@@ -512,20 +510,24 @@ func (p *PodFacts) genGatherScriptBase(vdb *vapi.VerticaDB, pf *PodFact) string
 	))
 }
 
-func (p *PodFacts) genGatherScriptForVerticaPIDCollection() string {
+func (p *PodFacts) genGatherScriptForVerticaPIDCollection(pf *PodFact) string {
 	// This is only be run when vertica process is running in the container we
 	// are doing the gather. And this is only true if we aren't running the nma
 	// in a separate container.
-	return dedent.Dedent(`
+	//
+	// The pgrep command is used to search for the Vertica process with the '-h
+	// <podIP>' parameter set. This is necessary because in some cases, when a
+	// pod is recreated, it may temporarily show the Vertica process from its
+	// previous incarnation. Since the pod's IP changes in such scenarios, using
+	// the '-h <podIP>' parameter ensures that we identify the Vertica process
+	// that started after the pod was created, adding an extra layer of
+	// fail-safe.
+	return dedent.Dedent(fmt.Sprintf(`
 		echo -n 'verticaPIDRunning: '
-		[[ $(pgrep -f "^.*vertica\s-D") ]] && echo true || echo false
+		[[ $(pgrep -f "/opt/vertica/bin/vertica.*-h %s") ]] && echo true || echo false
 		echo -n 'verticaProcess: '
 		pgrep -f "^.*vertica\s-D" -a | tail -1 || echo error
-		echo -n 'spreadPIDRunning: '
```

Comment:
Why were we checking spread pid?

Reply:
It was a prior round of debug code. I was curious to see if vertica and spread process both show up after a pod deletion.